lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -355,6 +359,14 @@ bool SPIRVRegularizeLLVM::regularize() {
               assert(Extract->user_empty());
               Extract->dropAllReferences();
               ToErase.push_back(Extract);
+            } else if (auto *Store = dyn_cast<StoreInst>(U)) {
```

Comment:
I'm okay with this fix, but I tend to think that the whole loop over `Cmpxchg->users()` should be removed - since we are operating here on LLVM IR, let's just always create desired composite and replace `cmpxchg` instruction with that composite. So, regardless of how it is used further the code will always be correct.

Reply:
Seems to be non-optimal(extra insert instructions) in some cases, but I agree that it should work, i.e. it is a more robust solution. Should we do it with another PR?

Reply:
> Should we do it with another PR?

I'm fine with implementing it in a separate PR

Reply:
Created https://github.com/KhronosGroup/SPIRV-LLVM-Translator/issues/866 to track it.